patches/minecraft/net/minecraft/world/level/Level.java.patch

```diff
@@ -17,13 +17,14 @@
     private final Thread f_46423_;
     private final boolean f_46424_;
     private int f_46425_;
-@@ -99,8 +_,12 @@
+@@ -99,8 +_,13 @@
     private final WorldBorder f_46447_;
     private final BiomeManager f_46420_;
     private final ResourceKey<Level> f_46421_;
 +   public boolean restoringBlockSnapshots = false;
 +   public boolean captureBlockSnapshots = false;
 +   public java.util.ArrayList<net.minecraftforge.common.util.BlockSnapshot> capturedBlockSnapshots = new java.util.ArrayList<>();
++   public java.util.Set<BlockEntity> freshBlockEntities = com.google.common.collect.Sets.newHashSet();
```

Comment:
Is a HashSet the right collection type here? It doesn't seem like the best choice based on how its being used, as it being a set vs a list doesn't seem to be necessary.

Reply:
The Set was just the first thing I thought off. I could make it an ArrayList instead if that is better (my knowledge about data structures is severely lacking to be completely honest).

Reply:
When I implemented a very similar system for my mod, I settled on an ArrayList for the iteration speed. In this use case, an ArrayList would probably use less memory and be faster. With an ArrayList an ensureCapacity call could also be made just ahead of adding everything to avoid reallocs in the add loop, though that would probably have less of an impact than just using an ArrayList in the first place.

Reply:
Changed it to an ArrayList and included your suggestion to call `ensureCapacity()` before adding the `BlockEntities` to it